src/measure_theory/weak_convergence.lean

```diff
@@ -0,0 +1,273 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.bochner_integration
+
+/-!
+# Weak convergence of finite Borel measures and Borel probability measures
+
+In this file we introduce `probability_measures α` and `finite_measures α` for any measurable
+space α. Both are defined as subtypes of measures on α. When the measurable space α is a
+topological space with its Borel sigma algebra, we define a topology on `probability_measures α`
+and on `finite_measures α`, which corresponds to the weak convergence of measures.
+
+## Main results
+
+- `weak_conv_seq_iff'`: A characterization of the weak convergence of a sequence of Borel
+  probability measures in terms of integrals of nonnegative bounded continuous functions.
+- WIP: Future versions will include many other equivalent characterizations of the weak convergence
+  of a sequence of Borel probability measures (Portmanteau theorem).
+
+## Notation
+
+ - No new notation is introduced.
+
+## References
+
+Weak convergence of measures in Wikipedia:
+<https://en.wikipedia.org/wiki/Convergence_of_measures#Weak_convergence_of_measures>
+
+A standard textbook about the weak convergence of Borel probability measures is [Billingsley1999].
+
+[Billingsley1999] Billingsley, Patrick (1999). Convergence of Probability Measures. New York, NY:
+John Wiley & Sons, Inc. ISBN 0-471-19745-9.
```

Comment:
Should this go in https://github.com/leanprover-community/mathlib/blob/master/docs/references.bib ?

Reply:
Yes! I thought touching only one file makes this PR look simpler, and this reference should be moved to its right place in the end. The other currently "incorrectly placed" piece is `def bounded_above`.

So I will move these two to their right places either in the end when it has been decided whether this PR will be accepted at all (there are probably major implementation issues to discuss before), or I could do that sooner, if it is seen as decluttering. For now I will not yet resolve this conversation, but keep it as a reminder to move these in the end.

Thank you for the comment/reminder!